### PR TITLE
[Merged by Bors] - chore: Homogenise instances for `MulOpposite`/`AddOpposite`

### PR DESCRIPTION
**Pull request by YaelDillies**
by declaring them all in `where` style with implicit type assumptions and `inst` prefix

Here to reduce the diff from #11203

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Field/Opposite.lean [hunk 1]
```diff
@@ -35,44 +33,46 @@ theorem unop_ratCast [RatCast α] (q : ℚ) : unop (q : αᵐᵒᵖ) = q :=
...
 
-variable (α)
+instance instDivisionSemiring [DivisionSemiring α] : DivisionSemiring αᵐᵒᵖ where
+  toSemiring := instSemiring
+  __ := instGroupWithZero
 
-instance instDivisionSemiring [DivisionSemiring α] : DivisionSemiring αᵐᵒᵖ :=
-  { MulOpposite.instGroupWithZero α, MulOpposite.instSemiring α with }
+instance instDivisionRing [DivisionRing α] : DivisionRing αᵐᵒᵖ where
+  toRing := instRing
```
Comment (eric-wieser) [inline Mathlib/Algebra/Field/Opposite.lean:41]:
```suggestion
  __ := instRing
```

**Comment by eric-wieser** [comment]
!bench

File: Mathlib/Algebra/Field/Opposite.lean (hunk 1 above)
Comment (eric-wieser) [reply Mathlib/Algebra/Field/Opposite.lean:41]: (and throughout)

Comment (YaelDillies) [reply Mathlib/Algebra/Field/Opposite.lean:41]: Why should I do this? It's more prone to people accidentally rearranging sources and messing up the instance.

Comment (eric-wieser) [reply Mathlib/Algebra/Field/Opposite.lean:41]: I think that's a problem for a linter to solve, not something that we want in the source code. This spelling is more brittle than using `__` everywhere, and still doesn't stop you writing `toTheGrandparentClass :=` and still making a mess.

Comment (eric-wieser) [reply Mathlib/Algebra/Field/Opposite.lean:41]: Also note that despite you "fixing the mess" of some of the parents orders, this seems not to have actually impacted performance.

Comment (YaelDillies) [reply Mathlib/Algebra/Field/Opposite.lean:41]: I suspect this is more due to the relatively little use of `MulOpposite` than poor performance of my technique

Comment (eric-wieser) [reply Mathlib/Algebra/Field/Opposite.lean:41]: Oh, I don't claim your technique is problematic performance-wise; I think it's uglier and more fragile than my suggested technique, while being exactly the same performance-wise.

**Comment by fpvandoorn** [comment]
bors merge
bors d+

**Pull request closed without merging**